docs/guides/cli/20_proxy-and-tunnel.mdx

```diff
@@ -0,0 +1,376 @@
+```
+
+### Running on Virtual and Dedicated Servers
+
+To use the Proxy with an application deployed on a VM or a Dedicated Server, 
+run the application server and the Proxy in parallel.
+
+For example, if you are running a NodeJS server on port `3000` and you expose web
+traffic on port `8080`: 
+
+```shell
+ory proxy --port 8080 http://127.0.0.1:3000 https://your-domain.com &
+
+# This should be your server command
+node your-entrypoint.js
+```
+
+### Running in Docker
+
+To use the Proxy in Docker, add your application and the Proxy to a single 
+Docker container. 
+
+:::warning
+
+This is not a perfect solution as Docker watches only the processes running in the foreground.
+With the application and the Proxy in one container, you must decide which becomes
+the foreground process. 
+
+When the processes crash, Docker notifies only of the failures of the
+foreground process and gives no information about the background process. 
+
+In this example, Ory Proxy is the background process. 
+
+Read the [Docker documentation](https://docs.docker.com/config/containers/multi-service_container/) to learn more about running multiple processes in a single container. 
+
+:::
+
+To run Ory Proxy in one container with your application, create an
+`entrypoint.sh` file similar to this:
+
+```shell
+#!/bin/bash
+set -e
+
+# This assumes that your server is running on port 3000. If it's running on a different port,
+# remember to adjust the command to point to the port where your server is running at!
+ory proxy --port 8080 http://127.0.0.1:3000 https://your-domain.com &
+
+# This should be your server command
+node your-entrypoint.js
+
+# Or, depending on the tech you use:
+#   go run main.go
+#   php index.php
+#   ...
+```
+
+Then, add the script to your Dockerfile and run it:
+
+```Dockerfile
+# ...
+
+EXPOSE 8080
+
+COPY ./docker-entrypoint.sh /
+ENTRYPOINT ["/docker-entrypoint.sh"]
+```
+
+## Ory Tunnel
+
+The Ory Tunnel has the same funtionality and behaves the same way 
+as the Proxy with these exceptions:  
+
+  ❌ Ory Tunnel doesn't alter incoming HTTP headers <br />
+  ❌ Ory Tunnel doesn't convert sessions into JSON Web Tokens, doesn't issue JTWs, 
+  and does't use JWTs natively <br />
+  ❌ Add the `.ory` prefix to any of the Ory URLs
+
+To run the Tunnel, use the `ory tunnel` command and point to the URL with your application, for example: 
+
+```shell
+ory tunnel http://localhost:3000
+```
+
+### Usage
+
+You can use Ory Tunnel the same way you can use Ory Proxy. The tool can be used for local development, 
+in production, on different server types, and in Docker. 
+
+For more details, refer to the [Ory Proxy documentation](#ory-proxy).
+
+## URL Structure
+
+:::tip
+Ory Proxy adds `/.ory` prefix when mirroring the Ory Cloud APIs and UIs. 
+:::
+
+For example, when using the Ory Proxy, calling 
+`https://<proxy-host>/.ory/ui/login` is the same as calling `https://<your-project-slug>.projects.oryapis.com/ui/login`
+directly.
+
```

Comment:
why are we talking about the Ory proxy again here?

Reply:
This is something I ported over from the old doc: 
https://www.ory.sh/docs/guides/proxy#url-structure

What's wrong with this section?

Reply:
since we already spoke about the proxy in the previous section this might be a bit confusing here? Maybe move this up to the `proxy` section?

Reply:
Still, if you look at the markdown table, there's a version that shows how the self-service URLs work in Tunnel (no prefix) and that's why I made it a separate section - it applies to both the Proxy and the Tunnel. But I guess it doesn't make much sense to include that? I mean, the tunnel doesn't alter the default self-service URLs?

Reply:
yeah the tunnel doesn't alter it. i think you can though through the a flag

Reply:
Okay, I moved this section under the Proxy and removed the switches to toggle between the proxy and the tunnel since the tunnel doesn't affect the regular self-service URLs.